ding/utils/log_writer_helper.py

```diff
@@ -0,0 +1,97 @@
+from tensorboardX import SummaryWriter
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from ding.framework import Task
+
+
+class DistributedWriter(SummaryWriter):
+    """
+    Overview:
+        A simple subclass of SummaryWriter that supports writing to one process in multi-process mode.
+        The best way is to use it in conjunction with the ``task`` to take advantage of the message \
+            and event components of the task (see ``writer.plugin``).
+    """
+
+    def __init__(self, *args, **kwargs):
+        self._default_writer_to_disk = kwargs.get("write_to_disk") if "write_to_disk" in kwargs else True
+        kwargs["write_to_disk"] = False
+        super().__init__(*args, **kwargs)
+        self._in_parallel = False
+        self._task = None
+        self._is_writer = False
+        self._lazy_initialized = False
+
+    def plugin(self, task: "Task", is_writer: False) -> "DistributedWriter":
+        """
+        Overview:
+            Plugin ``task``, so when using this writer in the task pipeline, it will automatically send requests\
+                to the main writer instead of writing it to the disk. So we can collect data from multiple processes\
+                and write them into one file.
+        Usage:
```

Comment:
You can use `Example` comments like [this](https://github.com/opendilab/DI-engine/blob/main/ding/model/template/q_learning.py#L89)

Reply:
Done